src/handlers/AnchorHandler.ts

```diff
@@ -0,0 +1,76 @@
+import BigNumber from 'bignumber.js';
+import ContractEntityHandler from './ContractEntityHandler';
+import {
+  AuxiliaryChain,
+  AuxiliaryChainRepository,
+} from '../repositories/AuxiliaryChainRepository';
+import Logger from '../Logger';
+import { AuxiliaryChainRecordNotFoundException } from '../Exception';
+
+/**
+ * This class handles Anchor event
+ */
+export default class AnchorHandler extends ContractEntityHandler<AuxiliaryChain> {
+  private auxiliaryChainRepository: AuxiliaryChainRepository;
+
+  private auxiliaryChainID: number;
+
+  /**
+   * Constructor
+   * @param auxiliaryChainRepository Instance of auxiliary chain repository.
+   * @param auxiliaryChainID Auxiliary chain Id.
+   */
+  public constructor(auxiliaryChainRepository: AuxiliaryChainRepository, auxiliaryChainID: number) {
+    super();
+    this.auxiliaryChainRepository = auxiliaryChainRepository;
+    this.auxiliaryChainID = auxiliaryChainID;
+  }
+
+  /**
+   * This method update latest origin block height.
+   *
+   * @param transactions Bulk transactions.
+   */
+  public async persist(transactions: any[]): Promise<AuxiliaryChain[]> {
+    const chainRecord = await this.auxiliaryChainRepository.get(this.auxiliaryChainID);
+    if (chainRecord == null) {
+      throw new AuxiliaryChainRecordNotFoundException('Cannot find record for auxiliary chain');
+    }
+
+    let anchorBlockHeight = chainRecord.lastOriginBlockHeight;
+    transactions
+      .filter(transaction => chainRecord.anchorAddress === transaction.contractAddress)
+      .forEach((filteredTransaction) => {
+        if (
+          anchorBlockHeight === undefined
+          || anchorBlockHeight.lt(new BigNumber(filteredTransaction._blockHeight))
+        ) {
+          anchorBlockHeight = new BigNumber(filteredTransaction._blockHeight);
+        }
+      });
+
+    // No change in block height of interested anchor.
+    if (anchorBlockHeight === undefined) {
+      return [chainRecord];
+    }
+
+    const hasChanged = chainRecord.lastOriginBlockHeight === undefined
+      || !chainRecord.lastOriginBlockHeight.eq(anchorBlockHeight);
+
+    if (hasChanged) {
+      this.auxiliaryChainRepository.update(chainRecord);
```

Comment:
Maybe I am missing something, however, do we modify `chainRecord` after reading it from database?

Reply:
Yes, we are reading it and updating it again.

Reply:
I might still miss something, however, should we have the following before updating chainRecord:

chainRecord.lastOriginBlockHeight = anchorBlockHeight;

Reply:
Oh yes 😱  🤦‍♂ 
I asserted in my test that it should update the blockHeight. But the test was passing because it was updating the same reference. I fixed the test as well. Thanks.

Reply:
Thx for update :).